src/main/java/seedu/address/logic/commands/StackUndoRedo.java

```diff
@@ -10,48 +10,61 @@
 
 public class StackUndoRedo {
     private Stack<RedoableCommand> undoStack;
+    private Stack<String> commandUndoStack;
+
     private Stack<RedoableCommand> redoStack;
+    private Stack<String> commandRedoStack;
 
```

Comment:
Adding the two stacks as well as the ModelStack subclass seems a bit overcomplicated? In `LogicManager`, you set data of UndoStack into `command`. Can't you set `commandText` into `command` as well and just put a getter to use in `UndoCommand` and `RedoCommand`? I might be wrong tho, maybe you can elaborate on your reasoning.

Reply:
I added another stack to store in the command lines that were executed by the user. So everything is in sequence with the redoablecommand. I could simply use getClass() of the command to get the name but then it wont include the args that was stated by the user.